content/6_Advanced/SCC.mdx

```diff
@@ -258,24 +258,231 @@ int main() {
 
 ## 2-SAT
 
-<FocusProblem problem="satEx" />
-
-<IncompleteSection>
-
-implementation
-
-</IncompleteSection>
-
-### Tutorial
+### Resources
 
 <Resources>
 	<Resource source="CF" title="2-SAT" url="16205" />
 	<Resource source="cp-algo" title="2-SAT" url="https://cp-algorithms.com/graph/2SAT.html#algorithm" />
 	<Resource source="Algorithms Live!" title="2SAT" url="https://youtu.be/0nNYy3rltgA" />
 </Resources>
 
+## Introduction
+The 2-Satisfiability (2-SAT) problem is a specific type of [Boolean satisfiability problem](https://en.wikipedia.org/wiki/Boolean_satisfiability_problem) where each clause in the given formula contains exactly two literals. The task is to determine whether there exists an assignment of truth values to the variables such that the entire formula is satisfied. A literal is either a variable or its negation. The formula is represented in [Conjunctive Normal Form (CNF)](https://en.wikipedia.org/wiki/Conjunctive_normal_form), which means it is a conjunction (AND) of clauses, where each clause is a disjunction (OR) of exactly two literals.
+
+## Algorithm
+To solve a 2-SAT problem, consider a Boolean formula in Conjunctive Normal Form (CNF) where each clause contains exactly two literals, such as
+
+$$
+(\lnot x_1 \lor x_2) \land (x_1 \lor \lnot x_2) \land (\lnot x_2 \lor x_3)
+$$
+The algorithm proceeds by constructing an implication graph, a directed graph where each variable $x_i$ and its negation $\lnot x_i$ are represented as nodes. For each clause $(a \lor b)$, we add two directed edges: $\lnot a \rightarrow b$ and $\lnot b \rightarrow a$. These edges reflect the logical implications necessary to satisfy each clause.
```

Comment:
add a drawing as an example

Reply:
I added an image.... but I don't know if it is proper since I can't see the rendered page

Reply:
@SansPapyrus683  can you check this issue out..... or will it work fine

Reply:
uh it should be on the deployment

Reply:
So there is no way to know it beforehand. That's bad.

Reply:
i think we have a markdown editor here: https://usaco.guide/editor

Reply:
I do everything here only. 
I don't think you can check it unless the photo is already in the main repository.

Reply:
yeah fair enough